### PR TITLE
Simple BigMoney Agent

### PR DESCRIPTION
**Pull request by henry-prior**
If expected value of a hand is >= 8, buys max(Silver, Gold, Province), otherwise buys max(Silver, Gold). Always discards minimum coin value cards for Militia and trashes minimum coin value cards for Bandit.

### CODE REVIEW

File: engine/agent.py [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+    elif isinstance(decision, BuyPhaseDecision):
+        for idx, move in enumerate(decision.moves):
+            if move.card_name == card_name:
+                return [idx]
+    return [0]
+
+
+def get_minimum_coin_card(decision):
+    card_coins = [c.coins for c in decision.moves.card]
+    return [np.argmin(card_coins))]
```
Comment (benzyx) [inline engine/agent.py]: typo here, bracket mismatch

File: engine/agent.py [hunk 2]
```diff
@@ -85,4 +109,28 @@ def policy(self, decision, state_view):
...
+
+        if decision.prompt == 'Select a card to trash from enemy Bandit.' or \
+            decision.prompt == 'Discard down to 3 cards.':
+            return get_minimum_coin_card(decision)
+
+        if state_view.player.phase == TurnPhase.TREASURE_PHASE:
+            return [1]
+
+        if state_view.player.phase == TurnPhase.BUY_PHASE:
+            all_cards_money = [c.coins for c in state_view.player.all_cards]
```
Comment (benzyx) [inline engine/agent.py]: state_view.player gives a PlayerView object, which does not have all_cards as a member.

**Review by benzyx — CHANGES_REQUESTED** [review]
You can test your code by changing main.py to play StdinAgent against BigMoneyAgent

**Comment by henry-prior** [comment]
Agent is working now, I added the ability for the `PlayerView` to keep track of the previous deck state before the last shuffle, which I think is reasonable information.

**Comment by henry-prior** [comment]
I think there might be a bug in Bandit, I bought Bandit against BigMoney and played it and got the logging messages:
```
Player 1 plays a Bandit.
  Player 1 gains a Gold.
  Player 2 discards a Poacher.
  Player 2 discards a Estate.
```
I (Player 1) had a Poacher, but Player 2 definitely did not. Didn't have a chance to dig deeper but will do so when I get a chance.

**Comment by benzyx** [comment]
> I think there might be a bug in Bandit, I bought Bandit against BigMoney and played it and got the logging messages:
> 
> ```
> Player 1 plays a Bandit.
>   Player 1 gains a Gold.
>   Player 2 discards a Poacher.
>   Player 2 discards a Estate.
> ```
> 
> I (Player 1) had a Poacher, but Player 2 definitely did not. Didn't have a chance to dig deeper but will do so when I get a chance.

Really good catch!

I drew 2 cards from the player instead of the opponent in the bandit attack. I'm hotfixing it in a separate commit.

**Comment by benzyx** [comment]
LGTM

**Review by benzyx — APPROVED** [review]

**Pull request merged**